plutus-playground-client/test/Data/Array/ExtraTests.purs

```diff
@@ -49,3 +53,23 @@ moveTests = do
       equal
         [ 0, 3, 1, 2, 4 ]
         (move 3 1 [ 0, 1, 2, 3, 4 ])
+
+collapseTests :: TestSuite
+collapseTests = do
+  suite "collapse" do
+    test "Empty." do
+      equal
+        (collapse ([] :: Array (Map Boolean String)))
+        []
+    test "Concrete example." do
+      equal
+        ( collapse
+            [ Map.fromFoldable [ "Foo" /\ true, "Bar" /\ false ]
+            , Map.fromFoldable [ "Quux" /\ true, "Loop" /\ true ]
+            ]
+        )
+        [ 0 /\ "Bar" /\ false
+        , 0 /\ "Foo" /\ true
+        , 1 /\ "Loop" /\ true
+        , 1 /\ "Quux" /\ true
+        ]
```

Comment:
what happens with duplicate keys?

Reply:
That's really a question for the caller.

We only ask for `Foldable`s on the input. If you supply a `Map`, then your choice of map determines how duplicate keys are handled. `collapse` doesn't care.